build/android.mk

```diff
@@ -209,29 +213,46 @@ ifeq ($(FAT_BINARY),y)
 ANDROID_LIB_BUILD =
 ANDROID_THIRDPARTY_STAMPS =
 
-# Example: $(eval $(call generate-abi,xcsoar,armeabi-v7a,ANDROID7))
+# Example: $(eval $(call generate-abi,xcsoar,armeabi-v7a,ANDROID7,n))
+# ("n" sets ANDROID_LEGACY)
 define generate-abi
 
 ANDROID_LIB_BUILD += $$(ANDROID_BUILD)/lib/$(2)/lib$(1).so
 
-$$(ANDROID_BUILD)/lib/$(2)/lib$(1).so: $$(OUT)/$(3)/$$(XCSOAR_ABI)/bin/lib$(1).so | $$(ANDROID_BUILD)/lib/$(2)/dirstamp
+# Legacy lib or not?
+ifeq ($(4),n)
+  ANDROID_THIRDPARTY_STAMP := $$(OUT)/$(3)/lib/$(2)/stamp
+  ANDROID_ABI_LIB := $$(OUT)/$(3)/$$(XCSOAR_ABI)/bin/lib$(1).so
+else
+  ANDROID_THIRDPARTY_STAMP := $$(OUT)/$(3)/lib_legacy/$(2)/stamp
+  ANDROID_ABI_LIB := $$(OUT)/$(3)/$$(XCSOAR_ABI)_legacy/bin/lib$(1).so
```

Comment:
For the legacy build, `XCSOAR_ABI` already contains the `_legacy` suffix. So what this does is add it again!